SPECS/gcc/gcc.spec

```diff
@@ -136,35 +219,139 @@ Requires:       libgomp = %{version}-%{release}
+building kernels for other architectures.  No support for cross-building
+user space programs is currently supplied as that would massively multiply the
+number of packages.
+%endif
+
+%do_package aarch64-linux-gnu %{build_cross}
+
 %prep
 %autosetup -p1
+
+function prep_target () {
+    local target=$1
+    local condition=$2
+
+    if [ $condition != 0 ]
+    then
+        echo $1 >> cross.list
+    fi
+}
+
+touch cross.list
+prep_target aarch64-linux-gnu %{build_cross}
+
 # disable no-pie for gcc binaries
 sed -i '/^NO_PIE_CFLAGS = /s/@NO_PIE_CFLAGS@//' gcc/Makefile.in
 
 %build
+function config_cross_target () {
+    local target=$1
+
+    mkdir $target
+    pushd $target
+
+    CFLAGS_FOR_TARGET="-g -O2 -Wall -fexceptions" \
+    AR_FOR_TARGET=%{_bindir}/$target-ar \
+    AS_FOR_TARGET=%{_bindir}/$target-as \
+    LD_FOR_TARGET=%{_bindir}/$target-ld \
+    NM_FOR_TARGET=%{_bindir}/$target-nm \
+    OBJDUMP_FOR_TARGET=%{_bindir}/$target-objdump \
+    RANLIB_FOR_TARGET=%{_bindir}/$target-ranlib \
+    READELF_FOR_TARGET=%{_bindir}/$target-readelf \
+    STRIP_FOR_TARGET=%{_bindir}/$target-strip \
+    SED=sed %configure \
+        --disable-bootstrap \
+        --disable-decimal-float \
+        --disable-dependency-tracking \
+        --disable-gold \
+        --disable-libgcj \
+        --disable-libgomp \
+        --disable-libmpx \
+        --disable-libquadmath \
+        --disable-libssp \
+        --disable-libunwind-exceptions \
+        --disable-multilib \
+        --disable-shared \
+        --disable-silent-rules \
+        --disable-sjlj-exceptions \
+        --disable-threads \
+        --enable-plugin \
+        --enable-__cxa_atexit \
+        --enable-clocale=gnu \
+        --enable-default-pie \
+        --enable-languages=c,c++ \
+        --enable-linker-build-id \
+        --enable-targets=all \
+        --program-prefix=$target- \
+        --target=$target \
+        --with-ld=/usr/bin/$target-ld \
+        --with-newlib \
+        --with-sysroot=%{_prefix}/$target/sys-root \
+        --with-system-zlib \
+        --without-headers
+
+    popd
+}
+
 CFLAGS="`echo " %{build_cflags} " | sed 's/-Werror=format-security/-Wno-error=format-security/'`"
 CXXFLAGS="`echo " %{build_cxxflags} " | sed 's/-Werror=format-security/-Wno-error=format-security/'`"
 FCFLAGS="`echo " %{build_fflags} " | sed 's/-Werror=format-security/-Wno-error=format-security/'`"
 export CFLAGS
 export CXXFLAGS
 export FCFLAGS
 
+mkdir build
+pushd build
+
 SED=sed \
 %configure \
-    --enable-shared \
-    --enable-threads=posix \
+    --disable-bootstrap \
+    --disable-multilib \
     --enable-__cxa_atexit \
     --enable-clocale=gnu \
+    --enable-default-pie \
     --enable-languages=c,c++,fortran \
-    --disable-multilib \
-    --disable-bootstrap \
     --enable-linker-build-id \
     --enable-plugin \
-    --enable-default-pie \
+    --enable-shared \
+    --enable-threads=posix \
     --with-system-zlib
```

Comment:
No changes to the default configuration - just putting the arguments in alphabetical order whenever possible.:)